### PR TITLE
More code refactoring and sync from upstream

### PR DESCRIPTION
**Pull request by whitneywhtsang**

### CODE REVIEW

File: third_party/intel/lib/TritonIntelGPUToLLVM/ReduceOpToLLVM.cpp [hunk 1]
```diff
@@ -80,35 +80,14 @@ struct ReduceOpConversion
 
   void accumulate(Location loc, ConversionPatternRewriter &rewriter,
                   Region &combineOp, SmallVector<Value> &acc, ValueRange cur,
-                  bool isFirst) const {
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUToLLVM/ReduceOpToLLVM.cpp:83]: Came from https://github.com/intel/intel-xpu-backend-for-triton/commit/1b0f9ea7f1f6e9961251d3a6ad97fd72521a59e5.

File: third_party/intel/lib/TritonIntelGPUToLLVM/ReduceOpToLLVM.cpp [hunk 2]
```diff
@@ -242,7 +231,7 @@ struct ReduceOpConversion
                     ConversionPatternRewriter &rewriter) const {
     auto srcLayout = helper.getSrcLayout();
     auto srcShape = helper.getSrcShape();
-    auto order = getOrder(srcLayout);
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUToLLVM/ReduceOpToLLVM.cpp:245]: Came from https://github.com/intel/intel-xpu-backend-for-triton/commit/53bfa70a71a41fe61c8275479a03f7f4b281d80c.

File: third_party/intel/lib/TritonIntelGPUToLLVM/ReduceScanCommon.h [hunk 3]
```diff
@@ -31,6 +31,91 @@ using ::mlir::triton::gpu::intel::DpasEncodingAttr;
 namespace mlir::triton {
 class ReduceOp;
 class ScanOp;
+
+inline SmallVector<Value>
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUToLLVM/ReduceScanCommon.h:35]: Came from https://github.com/intel/intel-xpu-backend-for-triton/commit/1b0f9ea7f1f6e9961251d3a6ad97fd72521a59e5.

File: third_party/intel/lib/TritonIntelGPUToLLVM/ReduceOpToLLVM.cpp [hunk 4]
```diff
@@ -284,7 +274,7 @@ struct ReduceOpConversion
 
     auto threadsPerWarp =
         triton::gpu::getThreadsPerWarpWithUniqueData(srcLayout, srcShape);
-    auto order = getOrder(srcLayout);
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUToLLVM/ReduceOpToLLVM.cpp:287]: Came from https://github.com/intel/intel-xpu-backend-for-triton/commit/755077cde95db74da41f3769293f83521699cabe.

**Review by pbchekin — APPROVED** [review]

**Review by etiotto — APPROVED** [review]

**Pull request merged**